src/components/Profile/index.js

```diff
@@ -0,0 +1,118 @@
+import {useState} from 'react';
+
+import {useStore} from '../../hooks/useStore';
+import Button from '../Button';
+import Dialog from '../Dialog';
+import Form from '../Form';
+
+import StyledProfile from './styled';
+
+export default function ProfileComponent() {
+	const users = useStore(state => state.users);
+	const activeUser = useStore(state => state.activeUser);
+	const setActiveUser = useStore(state => state.setActiveUser);
+	const changeUsername = useStore(state => state.changeUsername);
+	const [toggle, setToggle] = useState(true);
+	const [open, setOpen] = useState(false);
+
+	function changeActiveUser() {
+		users.map(user => {
+			if (user !== activeUser) {
+				setActiveUser(user);
+			}
+			return user;
+		});
+	}
+	const setNewUsername = event => {
+		event.preventDefault();
+		changeUsername(activeUser.userID, event.target.inputField.value);
+		event.target.reset();
+	};
+
+	return (
+		<StyledProfile>
+			<label>
+				<Button greyButton={true} onClick={changeActiveUser}>
+					Switch user
+				</Button>
+			</label>
+			<article id="userInformation">
+				{toggle ? (
+					<h1>{'Username: ' + activeUser.userName}</h1>
+				) : (
+					<Form
+						onSubmit={event => {
+							setNewUsername(event);
+							setToggle(!toggle);
+							setOpen(!open);
+							setTimeout(() => {
+								setOpen(false);
+							}, 1000);
+						}}
```

Comment:
To keep the JSX a bit cleaner, I would write a function `handleSubmit` and do everything within that function instead.
```suggestion
	const handleSubmit = event => {
		event.preventDefault();
		changeUsername(activeUser.userID, event.target.inputField.value);

		setToggle(!toggle);
		setOpen(!open);
		setTimeout(() => {
			setOpen(false);
		}, 1000);
		
	   event.target.reset();
	};

	return (
		<StyledProfile>
			<label>
				<Button greyButton={true} onClick={changeActiveUser}>
					Switch user
				</Button>
			</label>
			<article id="userInformation">
				{toggle ? (
					<h1>{'Username: ' + activeUser.userName}</h1>
				) : (
					<Form
						onSubmit={handleSubmit}
```

Reply:
Thanks for the hint! i´ve never thought of something like that. Will be done 😄